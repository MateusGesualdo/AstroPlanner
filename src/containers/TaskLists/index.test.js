import React, {useEffect} from 'react'
import renderer from 'react-test-renderer'
import {shallow} from 'enzyme'
import { TaskLists } from './index'

describe("Tasklists", () => {

    test("Snapshot", () => {
        
        const list = [{
            day: "Segunda",
            text: "text"
        }]
        const tree = renderer.create(<TaskLists taskList={list}/>).toJSON()

        expect(tree).toMatchSnapshot()
    })

    test("Deve buscar as tarefas no backend", () => {
        const fetchTasks = jest.fn()
        const list = [{
            day: "day",
            text: "text"
        }]
        const component = shallow(<TaskLists taskList={list} fetchTasks={fetchTasks}/> )

        component.simulate('load')

        expect(fetchTasks).toHaveBeenCalled()
    })

})