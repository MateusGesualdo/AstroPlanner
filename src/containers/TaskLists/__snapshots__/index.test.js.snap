// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tasklists Snapshot 1`] = `
<div
  className="sc-bdVaJa frqUrk"
>
  <div
    className="sc-bwzfXH hHiqeQ"
  >
    <h2>
      Segunda
    </h2>
    <ul>
      <li
        className="sc-htpNat kZXTHZ"
      >
        text
      </li>
    </ul>
  </div>
  <div
    className="sc-bwzfXH hHiqeQ"
  >
    <h2>
      Terça
    </h2>
    <ul />
  </div>
  <div
    className="sc-bwzfXH hHiqeQ"
  >
    <h2>
      Quarta
    </h2>
    <ul />
  </div>
  <div
    className="sc-bwzfXH hHiqeQ"
  >
    <h2>
      Quinta
    </h2>
    <ul />
  </div>
  <div
    className="sc-bwzfXH hHiqeQ"
  >
    <h2>
      Sexta
    </h2>
    <ul />
  </div>
  <div
    className="sc-bwzfXH hHiqeQ"
  >
    <h2>
      Sábado
    </h2>
    <ul />
  </div>
  <div
    className="sc-bwzfXH hHiqeQ"
  >
    <h2>
      Domingo
    </h2>
    <ul />
  </div>
</div>
`;
